assets/src/dashboard/utils/memoize.js

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export default function memoize(op, keyFromArgs = (args) => args.join('-')) {
+  const memoized = new Map();
+  return function (...args) {
+    const key = keyFromArgs(args);
+    return (
+      memoized.get(key) || (memoized.set(key, op(...args)), memoized.get(key))
+    );
+  };
+}
```

Comment:
need to test this, but quick memoization implementation to get us up and running